generative_ai/embeddings/document_retrieval_example.py

```diff
@@ -19,27 +19,41 @@
 
 
 def embed_text() -> list[list[float]]:
-    """Embeds texts with a pre-trained, foundational model.
+    """Calculates embeddings for the given text inputs.
 
     Returns:
-        A list of lists containing the embedding vectors for each input text
+        A list of lists containing the embedding vectors for each input text.
     """
 
-    # A list of texts to be embedded.
-    texts = ["banana muffins? ", "banana bread? banana muffins?"]
+    model = TextEmbeddingModel.from_pretrained("text-embedding-004")
+
+    inputs = [
+        # Option1: Use TextEmbeddingInput object as embedding input
+        TextEmbeddingInput(
+            text=(
+                "You'll find some apps on your Home screens, and all your apps in All Apps. "
+                "You can open apps, switch between apps, and find 2 apps at once."
+            ),
+            title="Find, open & close apps on a Pixel phone",
+            # The task type for embedding. Check the available tasks in the model's documentation.
+            task_type="RETRIEVAL_DOCUMENT"
+        ),
+        # Option2: Use plain text as embedding input
+        (
+            "You can download no-charge and paid apps from Google Play on your Android phone or tablet. "
+            "We recommend that you get apps from Google Play, but you can also get them from other sources."
+        )
+    ]
+
     # The dimensionality of the output embeddings.
-    dimensionality = 256
-    # The task type for embedding. Check the available tasks in the model's documentation.
-    task = "RETRIEVAL_DOCUMENT"
+    dimensionality = 128
```

Comment:
Reduce dimensionality to be mindful of the end user's wallet. The response `metadata` shows `{'billableCharacterCount': 296.0}`, which is ~ $0.00006 (which should be fine)